samples/tests/ComponentExamples/ComponentExamples.test.ts

```diff
@@ -22,35 +24,39 @@ test.describe('Component Examples Test', () => {
 
   // only need to make sure it is loading correctly, the rest will be the responsibility of composite tests
   test('Whether html page is loaded right with 2 composites', async ({ page }) => {
-    expect(await page.waitForSelector('text=Connect components to StatefulClient')).toBeTruthy();
-    expect(await page.waitForSelector('text=Render single video stream in your own component')).toBeTruthy();
     // Wait for message to finish sending
-    await page.waitForTimeout(1000);
+    await page.waitForSelector(dataUiId('sendbox-textfield'));
     await page.type(dataUiId('sendbox-textfield'), 'Hello!');
     await page.keyboard.press('Enter');
 
-    await page.waitForSelector(`[data-ui-status="delivered"]`);
+    await page.waitForSelector(dataUiStatus('delivered'));
 
     await page.addScriptTag({
       content: `document.querySelector('[data-ui-id=message-timestamp]').innerText='timestamp';`
     });
 
     // Start local video using custom start button
+
     await page.click(dataUiId('custom-start-button'));
     await page.waitForFunction(() => {
       const videoNode = document.querySelector('video');
-      return document.querySelectorAll('video').length === 1 && videoNode?.readyState === 4;
+      // HTMLMediaElement.HAVE_ENOUGH_DATA means media is ready for play
+      return (
+        document.querySelectorAll('video').length === 1 && videoNode?.readyState === HTMLMediaElement.HAVE_ENOUGH_DATA
+      );
     });
 
     expect(await page.screenshot({ fullPage: true })).toMatchSnapshot('componentExamples-localVideoStarted.png');
 
-    // Stop local video using CameraButton from UI Library
-    await page.click('id=camera-button');
+    // TODO: page.click on cameraButton is causing some flacky test issues(but manual click is fine)
```

Comment:
nit: flacky.

Also, it's better to figure this out now rather than add 5s delay to each test. Also, adding timeout can *still* cause flakiness on slow CI bots / probabilistically.

What flakiness do you see?

Reply:
When we click the camera button using .click(), it just doesn't work sometimes - but it works everytime when you manually click on it

Reply:
Finally get it fixed! it is related to call status, I add an element to track that